File: inflearn/java/2-01/Main.java
import java.util.*;

public class Main {
    public static String solution(int n, String[] arr) {
        String answer = arr[0] + " ";
        for (int i = 1; i < n; i++) {
            int a = Integer.parseInt(arr[i - 1]);
            int b = Integer.parseInt(arr[i]);
            if (a < b)
                answer += b + " ";
        }
        return answer;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        sc.nextLine(); // 입력 버퍼 클리어
        String[] input = sc.nextLine().split(" ");

        System.out.println(solution(n, input));
        sc.close();
    }
}

File: inflearn/javascript/2-01/main.js
const ps = require('../modules/ps');

const solution = (n, arr) => {
  let answer = arr[0] + ' '; // 첫 번째 수는 무조건 출력
  let big = arr[0];
  for (let i = 1; i < n; i++) {
    if (arr[i-1] < arr[i]) {
      answer += arr[i] + ' ';
    }
  }
  return answer;
};

ps.main(async () => {
  let n = parseInt(await ps.readLine()); // 문자열을 입력 받아 정수 반환
  let arr = await ps.readArrayLine(); // 문자열을 입력 받아 Number 배열 형으로 반환

  console.log(solution(n, arr));
});
